composables/tiptap/suggestion.ts

```diff
@@ -64,10 +64,10 @@ export const TiptapEmojiSuggestion: Partial<SuggestionOptions> = {
 
     const emojis = await import('@emoji-mart/data')
       .then(r => r.default as EmojiMartData)
-      .then(data => Object.values(data.emojis).filter(({ id }) => id.startsWith(query)))
+      .then(data => Object.values(data.emojis).filter(({ id }) => id.toLowerCase().startsWith(query.toLowerCase())))
 
     const customEmojis: CustomEmoji[] = currentCustomEmojis.value.emojis
-      .filter(emoji => emoji.shortcode.startsWith(query))
+      .filter(emoji => emoji.shortcode.toLowerCase().startsWith(query.toLowerCase()))
```

Comment:
We should cache `query.toLowerCase()` to avoid recomputing it during the loops.
But I think we should do the same with `shortcode` and `id`, and cache the result of `import('@emoji-mart/data')` with the data as we needed in lowercase. This will also speed up things

Reply:
OK, I removed multiple calls of `toLowerCase()` for `query`.

Also, I used `useAsyncData` to cache emojimart data import. Is this the right approach for caching?

Regarding making `id` and `shortcode` lowercase, there is an issue. If we lower the shortcode beforehand, we cannot show the original shortcode for users like the below.

![Screenshot from 2024-01-21 18-29-35](https://github.com/elk-zone/elk/assets/1425259/2a20de2e-f352-44cb-8a85-a6b6aef21ad4)

This is problematic since some custom emoji use camelcase naming like `:thisIsCamelCaseEmojiName:`. The lowercased name `:thisiscamelcaseemojiname:` is a bit difficult to read.

Maybe we can keep as is since I assume it's not so heavy computation to call `toLowerCase()` as many times as emoji counts.